builders/trailsBuilder.js

```diff
@@ -66,6 +66,38 @@ const queries = [
       WHERE (type = 'hike' OR type = 'horse' OR type = 'bike') AND name != ''
     `
   },
+  {
+    name: "trail-endpoints",
+    minZoom: 9,
+    maxZoom: 10,
+    query: `
+      WITH filtered_trails as (
+        SELECT geog::geometry as geom from trails
+        WHERE ST_Length(geog) > 3000 AND type = 'hike' OR type = 'horse' OR type = 'bike' OR
+        type = 'motorcycle' OR type = 'atv' AND name != ''
+      ), startpoints AS (
+        SELECT st_startpoint(geom) as geom FROM filtered_trails
+      ), endpoints AS (
+        SELECT st_endpoint(geom) as geom FROM filtered_trails
+      ) SELECT * FROM startpoints UNION ALL SELECT * FROM endpoints;
```

Comment:
One thing to keep in mind... Union statements require items in both sides of the joint to be in the same order.  You could run into unexpected issues here if the st_startpoint / st_endpoint doesn't populate a start point / end point correctly for some reason, resulting in an uneven number of rows or misordered data.

I'd consider using a proper join if you can.

Reply:
sounds good, but is there a good way to get something like that with a join? basically I just want a massive flat array of unstructured points, and this was an easy way to get it.

Reply:
There's a way, but we'd need some kind of unique identifier for each trail (which I still need to do). 
 Or we could use Name to do it, but there appears to be duplicates of name in the DB currently. 
 
What are the odds of what I described actually happening?  If it's negligible then maybe add this as a TODO for us to address later.

Reply:
I'll just add a todo. The potential impact of misordered data here is extremely low; the points are not bound to anything since they are basically a layer of unordered, unidentified dots.

If we wanted to add some interactivity to them: say highlight the dots on trail hover, or make them move or disappear when a handle is dragged, then we would need an ID. But that isn't necessary now I think.